### ISSUE TITLE
Shelly Duo support

### ISSUE THREAD

**Issue by Misiu** [StyraHem/ShellyForHASS#195]
As asked in https://github.com/StyraHem/ShellyForHASS/issues/139 I'm creating this issue to track adding support for Shelly Duo.
CoAP payload and `/settings` info will be ok for start?

**Comment by chemelli74** [comment]
Hi @Misiu, please share /status, /settings and CoAP payload.

Thank you,

Simone

**Comment by ameennihad** [comment]
Hi @chemelli74, I also need this feature, can you tell me how can I get /status, /settings and CoAP payload to share here?

**Comment by ameennihad** [comment]
Will this help?
shellies/ShellyBulbDuo-123456/light/0/status {"ison":true,"brightness":100,"white":1,"temp":2738}

shellies/ShellyBulbDuo-123456/light/0/command on
shellies/ShellyBulbDuo-123456/light/0/command off

shellies/ShellyBulbDuo-123456/light/0/set {"brightness": 30, "white": 45, "temp": 2738, "turn": "on"}

brightness: 1-100 (optional)
white: 1-100 (optional)
temp: 2700-6500

**Comment by Misiu** [comment]
`/status`

```
{
    "wifi_sta": {
        "connected": true,
        "ssid": "XXX",
        "ip": "XXX.XXX.XXX.XXX",
        "rssi": -59
    },
    "cloud": {
        "enabled": false,
        "connected": false
    },
    "mqtt": {
        "connected": false
    },
    "time": "20:43",
    "serial": 1,
    "has_update": false,
    "mac": "21F4AAD1650F",
    "lights": [
        {
            "ison": true,
            "brightness": 100,
            "white": 39,
            "temp": 4182
        }
    ],
    "meters": [
        {
            "power": 9.00,
            "is_valid": "true"
        }
    ],
    "update": {
        "status": "idle",
        "has_update": false,
        "new_version": "20200129-155730/master@a18bfaec",
        "old_version": "20200129-155730/master@a18bfaec"
    },
    "ram_total": 50656,
    "ram_free": 40608,
    "fs_size": 233681,
    "fs_free": 167417,
    "uptime": 54
}
```

`/settings`

```
{
    "device": {
        "type": "SHBDUO-1",
        "mac": "21F4AAD1650F",
        "hostname": "ShellyBulbDuo-D2590B",
        "num_outputs": 1
    },
    "wifi_ap": {
        "enabled": false,
        "ssid": "ShellyBulbDuo-D2590B",
        "key": ""
    },
    "wifi_sta": {
        "enabled": true,
        "ssid": "XXX",
        "ipv4_method": "dhcp",
        "ip": null,
        "gw": null,
        "mask": null,
        "dns": null
    },
    "wifi_sta1": {
        "enabled": false,
        "ssid": null,
        "ipv4_method": "dhcp",
        "ip": null,
        "gw": null,
        "mask": null,
        "dns": null
    },
    "mqtt": {
        "enable": false,
        "server": "XXX.XXX.XX.X:1883",
        "user": "",
        "id": "ShellyBulbDuo-D2590B",
        "reconnect_timeout_max": 60.000000,
        "reconnect_timeout_min": 2.000000,
        "clean_session": true,
        "keep_alive": 60,
        "max_qos": 0,
        "retain": false,
        "update_period": 30
    },
    "coiot": {
        "update_period": 15
    },
    "sntp": {
        "server": "time.google.com"
    },
    "login": {
        "enabled": false,
        "unprotected": false,
        "username": "admin",
        "password": "admin"
    },
    "pin_code": "",
    "name": "",
    "fw": "20200129-155730/master@a18bfaec",
    "discoverable": true,
    "build_info": {
        "build_id": "20200129-155730/master@a18bfaec",
        "build_timestamp": "2020-01-29T15:57:30Z",
        "build_version": "1.0"
    },
    "cloud": {
        "enabled": false,
        "connected": false
    },
    "timezone": "Europe/Warsaw",
    "lat": 53.144001,
    "lng": 16.725599,
    "tzautodetect": true,
    "tz_utc_offset": 3600,
    "tz_dst": false,
    "tz_dst_auto": true,
    "time": "20:44",
    "hwinfo": {
        "hw_revision": "prod-2019-12",
        "batch_id": 0
    },
    "mode": "white",
    "transition": 1000,
    "lights": [
        {
            "ison": true,
            "brightness": 100,
            "white": 39,
            "temp": 4182,
            "default_state": "on",
            "auto_on": 0.00,
            "auto_off": 0.00,
            "schedule": false,
            "schedule_rules": []
        }
    ],
    "night_mode": {
        "enabled": 0,
        "start_time": "00:00",
        "end_time": "00:00",
        "brightness": 0
    }
}
```

**Comment by hakana** [comment]
Thanks, I just received some DUO so I can test myself now. There is a problem with the CoAP messages on DUO, so I'm waiting for a new to test with.

**Comment by Misiu** [comment]
@chemelli74 I was able to ad ShellyForHass using Configuration->Automations 🎉
Shelly Duo showed up on the Entitles list, but it took a while (I don't have the cloud enabled, maybe that's why).
I was able to add the entity to my view, I'm able to turn on/off the bulb, I can set the brightness, but setting color temperature doesn't work (at least for me)
When I turn off the bulb and move the brightness slider the device turns on (which is a good thing), and the toggle turns to on, but when I set the brightness to 0 the toggle doesn't turn off. Is this by design? This should be consistent.
Shelly Duo shows power consumption in the app. Can this be added to ShellyForHass? (I thinks it should show, but I'm unable to save settings in Integration setting panel)

After adding Shelly Duo to view it is shown as:
![image](https://user-images.githubusercontent.com/1741838/76039744-9c711a80-5f4d-11ea-9c06-bc2c682bd8ea.png)

When I click on the entity the bulb turns off, but the card needs long time to show as disabled, same when clicking on toggle:
![image](https://user-images.githubusercontent.com/1741838/76039808-d3dfc700-5f4d-11ea-98ac-a8f2a22f2d09.png)

I've changed the interval in the config to 10 seconds (via settings in integration), but those settings aren't getting updated. Should I create a separate issue for that?
